homeassistant/components/huawei_lte/__init__.py

```diff
@@ -165,10 +165,20 @@ def get_data(key: str, func: Callable[[None], Any]) -> None:
                 )
                 self.subscriptions.pop(key)
             except ResponseErrorLoginRequiredException:
-                _LOGGER.info(
-                    "%s requires authorization, excluding from future updates", key
-                )
-                self.subscriptions.pop(key)
+                if isinstance(self.connection, AuthorizedConnection):
+                    _LOGGER.debug("Trying to authorize again...")
+                    if self.connection.enforce_authorized_connection():
+                        _LOGGER.debug(
+                            "...success, %s will be updated by a future periodic run",
+                            key,
+                        )
+                    else:
+                        _LOGGER.debug("...failed")
+                else:
+                    _LOGGER.info(
+                        "%s requires authorization, excluding from future updates", key
+                    )
+                    self.subscriptions.pop(key)
```

Comment:
The first line of `get_data` uses square brackets to get the key out of subscriptions. By popping it here, it will start causing `KeyError` ?

Reply:
There's a `if not self.subscriptions[key]: return` guard in start of `get_data`.

Reply:
Right. So that will raise a KeyError.

Reply:
Ah, misread; it won't cause a `KeyError` because `self.subscriptions` is a `defaultdict(set)`, i.e. returns an empty set (falsey) for nonexistent keys.

Reply:
There's no particular reason for that either; converted to .get for clarity.